### ISSUE TITLE
Tree selection-type "ancestors" - Provide a valid label for form fields - (2036570417) 

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-design-system#5615]
### Module: 
 38 Tree and Tree Item 

### Violation: 
 Provide a valid label for form fields where the selection type is set to `"ancestors"`.

![image](https://user-images.githubusercontent.com/5023024/202574670-833d40c9-fa2d-4912-9835-b873987c43de.png)

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- [3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036570417 

### Severity: 
 10 

### Description: 
 [Issue]
There are form controls with visual labels that are not programmatically associated with the control. Examples include:

- Child 1
- Child 2
- Child 3

[User Impact]
Screen reader users will have difficulty determining the purpose of these controls. Speech input users will have difficulty navigating to them.

[Code Reference] 
 ```
<div aria-checked="false" aria-label="" class="toggle" role="checkbox" tabindex="0"><svg class="check-svg" viewbox="0 0 16 16"><path d=""></path></svg></div>
 ``` 

### Note: 
 [Suggestion]
Provide a valid label for form fields. Remove aria-label attribute from ``` <div> ``` elements. The best way to do this is to use a ``` <label> ``` element that contains checkbox and its label.

[Compliant Code Example] 
 ```
<label class="checkbox-label">
  <calcite-checkbox class="checkbox" data-test-id="checkbox" guid="calcite-checkbox-ffffef30-cd34-98b7-5e82-99bbe9d57198" scale="m" calcite-hydrated="" checked="">
    <div aria-checked="false" class="toggle" role="checkbox" tabindex="0">
      <svg aria-hidden="true" class="check-svg" viewbox="0 0 16 16"><path d=""></path></svg>
    </div>
  </calcite-checkbox>
  Child 1
  <slot></slot>
</label>
 ```
 

### Media Type: 
 Forms 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
@ashetland @SkyeSeitz Seeking design perspective on the `tree` where the `selection-type` is `"ancestors"` and if a refactor is needed. This is listed as a critical a11y issue, however we believe the impact isn't as high due to its limited use.

Hoping to determine use cases, and how we can solve the issue without impacting other `selection-type`s on the `tree` component.

**Comment by geospatialem** [comment]
Reallocated to April to combine with other `tree` design issues.

**Comment by geospatialem** [comment]
Reallocating to the May release for design considerations.

**Comment by ashetland** [comment]
Designs for replacing the Checkbox component with icons are available in this [Figma file](https://www.figma.com/file/7L8xbf0m1E6n9m2I4CBK5v/Tree%3A-selection-type-ancestors-%5B5615%5D?type=design&node-id=501-13100&mode=dev).

**Comment by abp6318** [comment]
#### Note
Hello from Team Hub Discovery -- we came across this issue while working on updating the accessibility of our tree facet. I did a whole bunch of digging and puzzling, with my conclusion being we probably just need to add a single `div` or similar wrapping element around the checkbox and label.

I wrote up a duplicate ticket by mistake, but I'll move my findings to this comment for future corrective measures!

-----
#### Summary
Using aXe DevTools, Hub Discovery realized that the underlying issue for inaccessible names on calcite tree items is from within the calcite component itself. We are requesting an update to this, and expect it to be a small change as we have a list of options that also use checkboxes in a similar manner, and the only difference is a single `div` wrapping the checkbox and it's option label together within the overarching label.

#### Images related to issue
![Screenshot 2023-10-24 at 11 39 13 AM](https://github.com/Esri/calcite-design-system/assets/60486980/4df283d9-6130-42d8-819e-6937f5d761cc)
![Screenshot 2023-10-24 at 11 39 46 AM](https://github.com/Esri/calcite-design-system/assets/60486980/a023199a-c5cc-4977-9ea6-df053d077d35)

#### My attempt at explaining what I think the problem might be
calcite-tree-item's implementation causes this (even in code pens):
```javascript
<label class={CSS.checkboxLabel} key="checkbox-label">
<!-- missing div here -->
          <calcite-checkbox
            checked={this.selected}
            class={CSS.checkbox}
            data-test-id="checkbox"
            indeterminate={this.hasChildren && this.indeterminate}
            scale={this.scale}
            tabIndex={-1}
          />
          {defaultSlotNode}
<!-- missing /div here -->
</label>
```

Results in:
![image](https://github.com/Esri/calcite-design-system/assets/60486980/58bf308a-481e-4423-b81d-0731804fa9f3)

-----

#### Comparison
Over on Hub Discovery, we own a component that makes use of calcite components to create a list of options with checkboxes, but it doesn't use the `calcite-tree` or `calcite-tree-item`. It is called the `arcgis-hub-facet-options` and has a similar structure to the `calcite-tree-item`. See the two below:
- `arcgis-hub-facet-options`: https://github.com/ArcGIS/opendata-ui/blob/4fbde2f248e85b67c0a3ea78e7ee3985bd8b7d1b/packages/hub-components/src/components/arcgis-hub-facet-options/arcgis-hub-facet-options.tsx#L146C13-L156
- `calcite-tree-item`: https://github.com/Esri/calcite-design-system/blob/4a8106fb4726ab65778f8db2ca52d1133cc8835f/packages/calcite-components/src/components/tree-item/tree-item.tsx#L227-L237

Notice the difference is a missing `div.option-container`.

-----

#### Other relevant notes
- [The component](https://github.com/Esri/calcite-design-system/blob/4a8106fb4726ab65778f8db2ca52d1133cc8835f/packages/calcite-components/src/components/tree-item/tree-item.tsx#L227-L237)
- [using this checkbox, that has an aria-label](https://github.com/Esri/calcite-design-system/blob/4a8106fb4726ab65778f8db2ca52d1133cc8835f/packages/calcite-components/src/components/checkbox/checkbox.tsx#L262)
- [defaulting to the component label](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/utils/label.ts#L170), rather than the content that we would rather use. 

#### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/tree/

#### Reproduction Steps

1. Open aXe DevTools
2. Run page scan
3. Look for "ARIA toggle fields must have an accessible name" (labeled as a serious issue)

#### Calcite package

- [X] @esri/calcite-components

-----

### Expected Behavior

Our implementation of arcgis-hub-facet-options is this:

```javascript
return <calcite-label key={key}>
        <div class="option-container"> <!-- this is what we need to get added in calcite (I think) -->
          <calcite-checkbox
            checked={option.selected}
            id={option.key}
            name={option.label}
            onCalciteCheckboxChange={checkboxChangeHandler}
          />
          <span class="option">{option.label}</span>
        </div> 
</calcite-label>
```

results in:
![Screenshot 2023-10-24 at 11 41 59 AM](https://github.com/Esri/calcite-design-system/assets/60486980/dcbf581a-6e97-47a6-b2ab-7f977e801886)

**Comment by geospatialem** [comment]
When the `tree` has a `selectionMode` of `"ancestors"`, the `checkbox` should add the `$text$` (innerText) of the `tree-item` via an `aria-label` to provide sufficent context to AT:

Something similar to:
```tsx
<label class={CSS.checkboxLabel} key="checkbox-label">
  <calcite-checkbox
    aria-label={defaultSlotNode.$text$}
    checked={this.selected}
    class={CSS.checkbox}
    data-test-id="checkbox"
    indeterminate={this.hasChildren && this.indeterminate}
    scale={this.scale}
    tabIndex={-1}
  />
  {defaultSlotNode}
</label>
```

**Comment by thomas-hervey** [comment]
Hello,
Do we have an expected delivery timeline on this issue? An upcoming January 2024 release of ArcGIS Hub will expose this issue on our popular search route and it would be helpful to relay to government customers if/when we're expecting a fix. 

Thanks

**Comment by geospatialem** [comment]
> Hello, Do we have an expected delivery timeline on this issue? An upcoming January 2024 release of ArcGIS Hub will expose this issue on our popular search route and it would be helpful to relay to government customers if/when we're expecting a fix.
> 
> Thanks

@thomas-hervey Thanks for reaching out, just added the Hub label for Calcite team members to verify the impact across teams. We're targeting our March release for a fix.

**Comment by geospatialem** [comment]
Verified with JAWS and NVDA in `2.12.0-next.11` with https://codepen.io/geospatialem/pen/MWMOKWV
